File: solutions/problem_0522_longest_uncommon_subsequence_ii/solution.py
"""
Problem 522: Longest Uncommon Subsequence II
===========================================
Difficulty: Medium
Tags: Array, Hash Table, Two Pointers, String, Sorting
Pattern: Two Pointers on Sorted Array

Time Complexity:  O(n log n)
Space Complexity: O(1)
"""

class Solution:
    def findLUSlength(self, strs: List[str]) -> int:
        # Sort + two pointers - O(n log n) time
        strs.sort()
        left, right = 0, len(strs) - 1
        result = 0
        while left < right:
            curr_sum = strs[left] + strs[right]
            if curr_sum < strs if isinstance(strs, int) else 0:
                left += 1
            else:
                right -= 1
        return result

File: solutions/problem_0522_longest_uncommon_subsequence_ii/solution.cpp
/*
 * Problem 522: Longest Uncommon Subsequence II
 * ===========================================
 * Difficulty: Medium
 * Tags: Array, Hash Table, Two Pointers, String, Sorting
 * Pattern: Two Pointers on Sorted Array
 *
 * Time Complexity:  O(n log n)
 * Space Complexity: O(1)
 */

#include <algorithm>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int findLUSlength(vector<string>& strs) {
        // Sort + two pointers - O(n log n) time
        sort(strs.begin(), strs.end());
        int left = 0, right = strs.size() - 1;
        while (left < right) {
            int curr = strs[left] + strs[right];
            if (curr < strs) {
                left++;
            } else {
                right--;
            }
        }
        return 0;
    }
};
